scaleway/resource_iam_group_test.go

```diff
@@ -0,0 +1,126 @@
+package scaleway
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	iam "github.com/scaleway/scaleway-sdk-go/api/iam/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func init() {
+	resource.AddTestSweepers("scaleway_iam_group", &resource.Sweeper{
+		Name: "scaleway_iam_group",
+		F:    testSweepIamGroup,
+	})
+}
+
+func testSweepIamGroup(_ string) error {
+	return sweep(func(scwClient *scw.Client) error {
+		api := iam.NewAPI(scwClient)
+
+		// Requiring organization_id in list request is temporary
+		organizationID := os.Getenv("DEFAULT_ORGANIZATION_ID")
+		listApps, err := api.ListGroups(&iam.ListGroupsRequest{
+			OrganizationID: &organizationID,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to list groups: %w", err)
+		}
+		for _, app := range listApps.Groups {
+			err = api.DeleteGroup(&iam.DeleteGroupRequest{
+				GroupID: app.ID,
+			})
+			if err != nil {
+				return fmt.Errorf("failed to delete group: %w", err)
+			}
+		}
+		return nil
+	})
+}
+
+func TestAccScalewayIamGroup_Basic(t *testing.T) {
+	SkipBetaTest(t)
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayIamGroupDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
```

Comment:
Add the following test
- With description and name
- Update Description
- Update name
- With UserIDs
- Update UserIDs
- RemoveUserIDs
- ApplicationIDs
- UpdateApplicationIDs
- RemoveApplicationIDs